test/jdk/java/util/IdentityHashMap/DefaultRemoveReplace.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.util.IdentityHashMap;
+import java.util.Objects;
+
+/*
+ * @test
+ * @bug 8178355
+ * @summary Use identity-based comparison for IdentityHashMap#remove and #replace
+ */
+public class DefaultRemoveReplace {
+    public static void main(String[] args) {
+        final String key = "key";
+        final String internedValue = "value";
+        final String constructedValue = new String(new char[]{'v', 'a', 'l', 'u', 'e'});
```

Comment:
Using:
```suggestion
        final String constructedValue = new String(internedValue);
```
will allow the internal `String.value` array to be shared: <https://github.com/openjdk/jdk/blob/bdf8a2a2050393e91800786f8d5a5d6805f936eb/src/java.base/share/classes/java/lang/String.java#L259-L265>

Reply:
I know. This is just to get rid of intellij idea's warning.